// framework7-vue/framework7-vue-bundle.js
import Framework7Vue from './shared/plugin.js';
import { f7, f7ready, theme } from './shared/f7.js';
import { useStore } from './shared/use-store.js';

import f7AccordionContent from './components/accordion-content.js';
import f7AccordionItem from './components/accordion-item.js';
import f7AccordionToggle from './components/accordion-toggle.js';
import f7Accordion from './components/accordion.js';
import f7ActionsButton from './components/actions-button.js';
import f7ActionsGroup from './components/actions-group.js';
import f7ActionsLabel from './components/actions-label.js';
import f7Actions from './components/actions.js';
import f7App from './components/app.js';
import f7AreaChart from './components/area-chart.js';
import f7Badge from './components/badge.js';
import f7BlockFooter from './components/block-footer.js';
import f7BlockHeader from './components/block-header.js';
import f7BlockTitle from './components/block-title.js';
import f7Block from './components/block.js';
import f7BreadcrumbsCollapsed from './components/breadcrumbs-collapsed.js';
import f7BreadcrumbsItem from './components/breadcrumbs-item.js';
import f7BreadcrumbsSeparator from './components/breadcrumbs-separator.js';
import f7Breadcrumbs from './components/breadcrumbs.js';
import f7Button from './components/button.js';
import f7CardContent from './components/card-content.js';
import f7CardFooter from './components/card-footer.js';
import f7CardHeader from './components/card-header.js';
import f7Card from './components/card.js';
import f7Checkbox from './components/checkbox.js';
import f7Chip from './components/chip.js';
import f7FabBackdrop from './components/fab-backdrop.js';
import f7FabButton from './components/fab-button.js';
import f7FabButtons from './components/fab-buttons.js';
import f7Fab from './components/fab.js';
import f7Gauge from './components/gauge.js';
import f7Icon from './components/icon.js';
import f7Input from './components/input.js';
import f7Link from './components/link.js';
import f7ListButton from './components/list-button.js';
import f7ListGroup from './components/list-group.js';
import f7ListIndex from './components/list-index.js';
import f7ListInput from './components/list-input.js';
import f7ListItem from './components/list-item.js';
import f7List from './components/list.js';
import f7LoginScreenTitle from './components/login-screen-title.js';
import f7LoginScreen from './components/login-screen.js';
import f7Message from './components/message.js';
import f7MessagebarAttachment from './components/messagebar-attachment.js';
import f7MessagebarAttachments from './components/messagebar-attachments.js';
import f7MessagebarSheetImage from './components/messagebar-sheet-image.js';
import f7MessagebarSheetItem from './components/messagebar-sheet-item.js';
import f7MessagebarSheet from './components/messagebar-sheet.js';
import f7Messagebar from './components/messagebar.js';
import f7MessagesTitle from './components/messages-title.js';
import f7Messages from './components/messages.js';
import f7NavLeft from './components/nav-left.js';
import f7NavRight from './components/nav-right.js';
import f7NavTitleLarge from './components/nav-title-large.js';
import f7NavTitle from './components/nav-title.js';
import f7Navbar from './components/navbar.js';
import f7PageContent from './components/page-content.js';
import f7Page from './components/page.js';
import f7Panel from './components/panel.js';
import f7PhotoBrowser from './components/photo-browser.js';
import f7PieChart from './components/pie-chart.js';
import f7Popover from './components/popover.js';
import f7Popup from './components/popup.js';
import f7Preloader from './components/preloader.js';
import f7Progressbar from './components/progressbar.js';
import f7Radio from './components/radio.js';
import f7Range from './components/range.js';
import f7RoutableModals from './components/routable-modals.js';
import f7Searchbar from './components/searchbar.js';
import f7Segmented from './components/segmented.js';
import f7Sheet from './components/sheet.js';
import f7SkeletonAvatar from './components/skeleton-avatar.js';
import f7SkeletonBlock from './components/skeleton-block.js';
import f7SkeletonImage from './components/skeleton-image.js';
import f7SkeletonText from './components/skeleton-text.js';
import f7Stepper from './components/stepper.js';
import f7Subnavbar from './components/subnavbar.js';
import f7SwipeoutActions from './components/swipeout-actions.js';
import f7SwipeoutButton from './components/swipeout-button.js';
import f7Tab from './components/tab.js';
import f7Tabs from './components/tabs.js';
import f7TextEditor from './components/text-editor.js';
import f7Toggle from './components/toggle.js';
import f7Toolbar from './components/toolbar.js';
import f7TreeviewItem from './components/treeview-item.js';
import f7Treeview from './components/treeview.js';
import f7UseIcon from './components/use-icon.js';
import f7View from './components/view.js';
import f7Views from './components/views.js';

export { f7AccordionContent, f7AccordionItem, f7AccordionToggle, f7Accordion, f7ActionsButton, f7ActionsGroup, f7ActionsLabel, f7Actions, f7App, f7AreaChart, f7Badge, f7BlockFooter, f7BlockHeader, f7BlockTitle, f7Block, f7BreadcrumbsCollapsed, f7BreadcrumbsItem, f7BreadcrumbsSeparator, f7Breadcrumbs, f7Button, f7CardContent, f7CardFooter, f7CardHeader, f7Card, f7Checkbox, f7Chip, f7FabBackdrop, f7FabButton, f7FabButtons, f7Fab, f7Gauge, f7Icon, f7Input, f7Link, f7ListButton, f7ListGroup, f7ListIndex, f7ListInput, f7ListItem, f7List, f7LoginScreenTitle, f7LoginScreen, f7Message, f7MessagebarAttachment, f7MessagebarAttachments, f7MessagebarSheetImage, f7MessagebarSheetItem, f7MessagebarSheet, f7Messagebar, f7MessagesTitle, f7Messages, f7NavLeft, f7NavRight, f7NavTitleLarge, f7NavTitle, f7Navbar, f7PageContent, f7Page, f7Panel, f7PhotoBrowser, f7PieChart, f7Popover, f7Popup, f7Preloader, f7Progressbar, f7Radio, f7Range, f7RoutableModals, f7Searchbar, f7Segmented, f7Sheet, f7SkeletonAvatar, f7SkeletonBlock, f7SkeletonImage, f7SkeletonText, f7Stepper, f7Subnavbar, f7SwipeoutActions, f7SwipeoutButton, f7Tab, f7Tabs, f7TextEditor, f7Toggle, f7Toolbar, f7TreeviewItem, f7Treeview, f7UseIcon, f7View, f7Views }
export { f7, f7ready, theme, useStore };
export default Framework7Vue;

function registerComponents(app) {
  app.component('f7-accordion-content', f7AccordionContent);
  app.component('f7-accordion-item', f7AccordionItem);
  app.component('f7-accordion-toggle', f7AccordionToggle);
  app.component('f7-accordion', f7Accordion);
  app.component('f7-actions-button', f7ActionsButton);
  app.component('f7-actions-group', f7ActionsGroup);
  app.component('f7-actions-label', f7ActionsLabel);
  app.component('f7-actions', f7Actions);
  app.component('f7-app', f7App);
  app.component('f7-area-chart', f7AreaChart);
  app.component('f7-badge', f7Badge);
  app.component('f7-block-footer', f7BlockFooter);
  app.component('f7-block-header', f7BlockHeader);
  app.component('f7-block-title', f7BlockTitle);
  app.component('f7-block', f7Block);
  app.component('f7-breadcrumbs-collapsed', f7BreadcrumbsCollapsed);
  app.component('f7-breadcrumbs-item', f7BreadcrumbsItem);
  app.component('f7-breadcrumbs-separator', f7BreadcrumbsSeparator);
  app.component('f7-breadcrumbs', f7Breadcrumbs);
  app.component('f7-button', f7Button);
  app.component('f7-card-content', f7CardContent);
  app.component('f7-card-footer', f7CardFooter);
  app.component('f7-card-header', f7CardHeader);
  app.component('f7-card', f7Card);
  app.component('f7-checkbox', f7Checkbox);
  app.component('f7-chip', f7Chip);
  app.component('f7-fab-backdrop', f7FabBackdrop);
  app.component('f7-fab-button', f7FabButton);
  app.component('f7-fab-buttons', f7FabButtons);
  app.component('f7-fab', f7Fab);
  app.component('f7-gauge', f7Gauge);
  app.component('f7-icon', f7Icon);
  app.component('f7-input', f7Input);
  app.component('f7-link', f7Link);
  app.component('f7-list-button', f7ListButton);
  app.component('f7-list-group', f7ListGroup);
  app.component('f7-list-index', f7ListIndex);
  app.component('f7-list-input', f7ListInput);
  app.component('f7-list-item', f7ListItem);
  app.component('f7-list', f7List);
  app.component('f7-login-screen-title', f7LoginScreenTitle);
  app.component('f7-login-screen', f7LoginScreen);
  app.component('f7-message', f7Message);
  app.component('f7-messagebar-attachment', f7MessagebarAttachment);
  app.component('f7-messagebar-attachments', f7MessagebarAttachments);
  app.component('f7-messagebar-sheet-image', f7MessagebarSheetImage);
  app.component('f7-messagebar-sheet-item', f7MessagebarSheetItem);
  app.component('f7-messagebar-sheet', f7MessagebarSheet);
  app.component('f7-messagebar', f7Messagebar);
  app.component('f7-messages-title', f7MessagesTitle);
  app.component('f7-messages', f7Messages);
  app.component('f7-nav-left', f7NavLeft);
  app.component('f7-nav-right', f7NavRight);
  app.component('f7-nav-title-large', f7NavTitleLarge);
  app.component('f7-nav-title', f7NavTitle);
  app.component('f7-navbar', f7Navbar);
  app.component('f7-page-content', f7PageContent);
  app.component('f7-page', f7Page);
  app.component('f7-panel', f7Panel);
  app.component('f7-photo-browser', f7PhotoBrowser);
  app.component('f7-pie-chart', f7PieChart);
  app.component('f7-popover', f7Popover);
  app.component('f7-popup', f7Popup);
  app.component('f7-preloader', f7Preloader);
  app.component('f7-progressbar', f7Progressbar);
  app.component('f7-radio', f7Radio);
  app.component('f7-range', f7Range);
  app.component('f7-routable-modals', f7RoutableModals);
  app.component('f7-searchbar', f7Searchbar);
  app.component('f7-segmented', f7Segmented);
  app.component('f7-sheet', f7Sheet);
  app.component('f7-skeleton-avatar', f7SkeletonAvatar);
  app.component('f7-skeleton-block', f7SkeletonBlock);
  app.component('f7-skeleton-image', f7SkeletonImage);
  app.component('f7-skeleton-text', f7SkeletonText);
  app.component('f7-stepper', f7Stepper);
  app.component('f7-subnavbar', f7Subnavbar);
  app.component('f7-swipeout-actions', f7SwipeoutActions);
  app.component('f7-swipeout-button', f7SwipeoutButton);
  app.component('f7-tab', f7Tab);
  app.component('f7-tabs', f7Tabs);
  app.component('f7-text-editor', f7TextEditor);
  app.component('f7-toggle', f7Toggle);
  app.component('f7-toolbar', f7Toolbar);
  app.component('f7-treeview-item', f7TreeviewItem);
  app.component('f7-treeview', f7Treeview);
  app.component('f7-use-icon', f7UseIcon);
  app.component('f7-view', f7View);
  app.component('f7-views', f7Views)
}
export { registerComponents }
