// framework7-vue/components/actions-label.js
import { renderSlot as _renderSlot, normalizeClass as _normalizeClass, openBlock as _openBlock, createElementBlock as _createElementBlock } from "vue";
function render(_ctx, _cache) {
  return _openBlock(), _createElementBlock("div", {
    class: _normalizeClass(_ctx.classes),
    onClick: _cache[0] || (_cache[0] = function () {
      return _ctx.onClick && _ctx.onClick(...arguments);
    })
  }, [_renderSlot(_ctx.$slots, "default")], 2);
}
import { computed } from 'vue';
import { classNames } from '../shared/utils.js';
import { colorClasses, colorProps } from '../shared/mixins.js';
export default {
  name: 'f7-actions-label',
  render,
  props: {
    strong: Boolean,
    ...colorProps
  },
  emits: ['click'],
  setup(props, _ref) {
    let {
      emit
    } = _ref;
    const onClick = e => {
      emit('click', e);
    };
    const classes = computed(() => classNames('actions-label', {
      'actions-button-strong': props.strong
    }, colorClasses(props)));
    return {
      classes,
      onClick
    };
  }
};
