// framework7-vue/components/nav-title.js
import { renderSlot as _renderSlot, toDisplayString as _toDisplayString, openBlock as _openBlock, createElementBlock as _createElementBlock, createCommentVNode as _createCommentVNode, createTextVNode as _createTextVNode, normalizeClass as _normalizeClass } from "vue";
const _hoisted_1 = {
  key: 0,
  class: "subtitle"
};
function render(_ctx, _cache) {
  return _openBlock(), _createElementBlock("div", {
    class: _normalizeClass(_ctx.classes)
  }, [_renderSlot(_ctx.$slots, "default"), _createTextVNode(" " + _toDisplayString(_ctx.title) + " ", 1), _ctx.subtitle ? (_openBlock(), _createElementBlock("span", _hoisted_1, _toDisplayString(_ctx.subtitle), 1)) : _createCommentVNode("", true)], 2);
}
import { computed } from 'vue';
import { classNames } from '../shared/utils.js';
import { colorClasses, colorProps } from '../shared/mixins.js';
export default {
  name: 'f7-nav-title',
  render,
  props: {
    title: String,
    subtitle: String,
    sliding: Boolean,
    ...colorProps
  },
  setup(props) {
    const classes = computed(() => classNames('title', {
      sliding: props.sliding
    }, colorClasses(props)));
    return {
      classes
    };
  }
};
