// framework7-vue/components/radio.js
import { createElementVNode as _createElementVNode, renderSlot as _renderSlot, normalizeClass as _normalizeClass, openBlock as _openBlock, createElementBlock as _createElementBlock } from "vue";
const _hoisted_1 = ["name", "value", "disabled", "readonly", "checked"];
const _hoisted_2 = /*#__PURE__*/_createElementVNode("i", {
  class: "icon-radio"
}, null, -1);
function render(_ctx, _cache) {
  return _openBlock(), _createElementBlock("label", {
    class: _normalizeClass(_ctx.classes)
  }, [_createElementVNode("input", {
    ref: "inputElRef",
    type: "radio",
    name: _ctx.name,
    value: _ctx.value,
    disabled: _ctx.disabled,
    readonly: _ctx.readonly,
    checked: _ctx.checked,
    onChange: _cache[0] || (_cache[0] = function () {
      return _ctx.onChange && _ctx.onChange(...arguments);
    })
  }, null, 40, _hoisted_1), _hoisted_2, _renderSlot(_ctx.$slots, "default")], 2);
}
import { ref, computed } from 'vue';
import { classNames } from '../shared/utils.js';
import { colorClasses, colorProps } from '../shared/mixins.js';
export default {
  name: 'f7-radio',
  render,
  props: {
    checked: Boolean,
    name: [Number, String],
    value: {
      type: [Number, String, Boolean],
      default: undefined
    },
    disabled: Boolean,
    readonly: Boolean,
    ...colorProps
  },
  emits: ['update:checked', 'change'],
  setup(props, _ref) {
    let {
      emit
    } = _ref;
    const inputElRef = ref(null);
    const onChange = event => {
      emit('update:checked', event.target.checked);
      emit('change', event);
    };
    const classes = computed(() => classNames({
      radio: true,
      disabled: props.disabled
    }, colorClasses(props)));
    return {
      inputElRef,
      classes,
      onChange
    };
  }
};
