// framework7-vue/components/subnavbar.js
import { toDisplayString as _toDisplayString, openBlock as _openBlock, createElementBlock as _createElementBlock, createCommentVNode as _createCommentVNode, renderSlot as _renderSlot, normalizeClass as _normalizeClass } from "vue";
const _hoisted_1 = {
  key: 0,
  class: "subnavbar-inner"
};
const _hoisted_2 = {
  key: 0,
  class: "subnavbar-title"
};
function render(_ctx, _cache) {
  return _openBlock(), _createElementBlock("div", {
    class: _normalizeClass(_ctx.classes)
  }, [_ctx.inner ? (_openBlock(), _createElementBlock("div", _hoisted_1, [_ctx.title ? (_openBlock(), _createElementBlock("div", _hoisted_2, _toDisplayString(_ctx.title), 1)) : _createCommentVNode("", true), _renderSlot(_ctx.$slots, "default")])) : _renderSlot(_ctx.$slots, "default", {
    key: 1
  })], 2);
}
import { computed } from 'vue';
import { classNames } from '../shared/utils.js';
import { colorClasses, colorProps } from '../shared/mixins.js';
export default {
  name: 'f7-subnavbar',
  render,
  props: {
    sliding: Boolean,
    title: String,
    inner: {
      type: Boolean,
      default: true
    },
    ...colorProps
  },
  setup(props) {
    const classes = computed(() => classNames('subnavbar', {
      sliding: props.sliding
    }, colorClasses(props)));
    return {
      classes
    };
  }
};
